content/doc/book/scaling/scaling-jenkins-on-kubernetes.adoc

```diff
@@ -1,15 +1,235 @@
+
+== Jenkins Scalability
+
+When Using the standalone Jenkins server you don’t have to worry about 
+multiple servers and nodes. 
+Howevere, the issue with this setup is that your server can become overloaded 
+with numerous jobs running at the same time. 
+There are ways to solve this problem by increasing the number of executors, 
+but you soon end up hitting a performance limit.
+
+To overcome this problem, you can offload some of the jobs to different machines 
+called Jenkins agents.
+
+Scalability is a measure that shows the ability of a system to expand its capabilities 
+to handle additional load. 
+One of the strongest sides of Jenkins is that it has a scaling feature almost out-of-the-box. Jenkins scaling is based on the controller/agents model, where you have a number of agent instances and one main Jenkins instance called the controller, which is responsible mainly for distributing jobs across the agents.
+Jenkins agents run a small program that communicates with the Jenkins controller to check if there’s any job it can run. When Jenkins finds a job scheduled, it transfers the build to the agent server.
+
+== Scaling Jenkins on top of Kubernetes
+
+Taking it a step further, running a container orchestrator such as Kubernetes, 
+you can make Jenkins do smarter things. You have the Jenkins controller who’s 
+responsible for the hosting configuration and the distribution of jobs to multiple agents. However, you don’t need the agents to be preexisting, they can be created on the fly, as in when they’re required.
+
+=== Advantages of scaling Jenkins on top of Kubernetes
+
+. *Autohealing is possible*: If your build or your agent gets corrupted, you no longer need to  worry — Jenkins will remove the unhealthy instance and spin up a new one.
+. *You can run your builds in parallel*: You no longer have to plan the executors and limit them; instead, Jenkins will spin up an agent instance and run your build in it.
+. *The load distribution is even*: Kubernetes manages loads well, and it’ll make sure your Jenkins agents spin up in the best available server, which makes your builds faster and more efficient.
+
+== Setting Up Scalable Jenkins on Kubernetes
+
+=== Jenkins Controller Installation
+
+To install a Jenkins controller, create your docker image based on the base Jenkins image, which can be found in the official docker repository (keep in mind that Docker should be installed on your local machine, see the installation steps link:/doc/book/installing/docker/#downloading-and-running-jenkins-in-docker[here]). 
+For this, you need to create a dockerfile. 
+Having your Jenkins setup as a dockerfile is highly recommended to make your continuous integration infrastructure replicable and have it ready for setup from scratch. 
+So let’s create our first example dockerfile for the Jenkins controller:
+
+.Dockerfile
+[source,text]
+----
+from jenkins/jenkins:lts
+
+# Distributed Builds plugins
+RUN /usr/local/bin/install-plugins.sh ssh-slaves
```

Comment:
This script is going to be deprecated soon, cc @timja @slide for confirmation. Please use the modern Plugin Installation Manager tool. Also, it would be great to use example without "slave"

Reply:
Yes, we definitely want to start recommending the plugin installation manager tool